### PR TITLE
Chore: update drand to v1.2.0

### PR DESCRIPTION
**Pull request by hsanjuan**
I also ran `go mod tidy`.

@willscott there should be no relevant changes to clients in this update right?

### CODE REVIEW

**Comment by willscott** [comment]
correct. this should have minimal impact / change. only 1.2 change in client is logging.

**Comment by willscott** [comment]
Lets get the [latest change](https://github.com/drand/drand/pull/759) from drand master into this (will release as v1.2.1 shortly).
That will let us safely remove [WithAutoWatch](https://github.com/filecoin-project/lotus/blob/master/chain/beacon/drand/drand.go#L95) and cause lotus to poll the drand http endpoints much less frequently.

**Comment by willscott** [comment]
Pushed a commit to this branch to update to v1.2.1

* Stops the 'autowatch' behavior
* sets user agent as `drand-client-lotus/<build version>`

File: chain/beacon/drand/drand.go [hunk 1]
```diff
@@ -84,6 +89,7 @@ func NewDrandBeacon(genesisTs, interval uint64, ps *pubsub.PubSub, config dtypes
 		if err != nil {
 			return nil, xerrors.Errorf("could not create http drand client: %w", err)
 		}
+		hc.(DrandHTTPClient).SetUserAgent("drand-client-lotus/" + build.BuildVersion)
```
Comment (hsanjuan) [inline chain/beacon/drand/drand.go:92]: Would it be possible to discriminate between miner/node etc. ?

Also, `CurrentCommit` might be better than BuildVersion (or both).

Comment (willscott) [reply chain/beacon/drand/drand.go:92]: i was worried that the distinct commit was a bit of a privacy invasive thing. There isn't a build constant for type of node, and i didn't see an easy way to ask lotus for that.

Comment (hsanjuan) [reply chain/beacon/drand/drand.go:92]: the `os.Executable` thing that would get used by default would give that right?

Comment (willscott) [reply chain/beacon/drand/drand.go:92]: maybe, but not reliably.
if the thing is being run via a `go run` then that doesn't work. Better to say lotus here, i think

Comment (hsanjuan) [reply chain/beacon/drand/drand.go:92]: Not that anyone is ever going to run lotus via `go run` I hope. But ok, eventually only the miners should be querying drand anyways.

Comment (Kubuxu) [reply chain/beacon/drand/drand.go:92]: Miner process always uses lotus daemon for getting its randomess.

**Review by Kubuxu — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

**Pull request merged**